fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -137,14 +140,19 @@ public synchronized void resetColumns(SegmentMetadata segmentMetadata, Dimension
             workingIntervals.clear();
             for (Map.Entry<String, Set<Interval>> nameIntervals : dimensionIntervals.entrySet()) {
                 String physicalName = nameIntervals.getKey();
-                String apiName = getLogicalColumnName(physicalName);
-                Dimension dimension = dimensionDictionary.findByApiName(apiName);
-                // Schema evolution may lead to unknown dimensions, skip these
-                if (dimension == null) {
-                    continue;
-                }
-                DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(this, dimension);
-                workingIntervals.put(dimensionColumn, nameIntervals.getValue());
+
+                getLogicalColumnNames(physicalName).stream()
+                        .map(dimensionDictionary::findByApiName)
+                        .filter(dimension -> dimension != null)
+                        .forEach(
+                                dimension -> {
+                                    DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(
```

Comment:
To elaborate on why it feels wrong to me, I generally prefer to be hit across the head with a big neon sign when an operation has side-effects, _especially_ when the operation returns a value as well. forEach does that, while a `map` operation does not.